corehq/apps/app_manager/models.py

```diff
@@ -2989,6 +2991,108 @@ def update_app_case_meta(self, app_case_meta):
                 meta.add_closer(self.unique_id, action.close_condition)
 
 
+class ShadowForm(AdvancedForm):
+    form_type = 'shadow_form'
+    # The unqiue id of the form we are shadowing
+    shadow_parent_form_id = StringProperty(required=False, default=None)
+
+    # form actions to be merged with the parent actions
+    extra_actions = SchemaProperty(AdvancedFormActions)
+
+    @property
+    def shadow_parent_form(self):
+        if not self.shadow_parent_form_id:
+            return None
+        else:
+            app = self.get_app()
+            return app.get_form(self.shadow_parent_form_id)
+
+    @property
+    def source(self):
+        if self.shadow_parent_form_id:
+            return self.shadow_parent_form.source
+        from corehq.apps.app_manager.views.utils import get_blank_form_xml
+        return get_blank_form_xml("Untitled Form")
+
+    @property
+    def xmlns(self):
+        if not self.shadow_parent_form_id:
+            return None
+        else:
+            return self.shadow_parent_form.xmlns
+
+    @property
+    def actions(self):
+        if not self.shadow_parent_form_id:
+            shadow_parent_actions = AdvancedFormActions()
+        else:
+            shadow_parent_actions = self.shadow_parent_form.actions
+        return self._merge_actions(shadow_parent_actions, self.extra_actions)
+
+    def get_shadow_parent_options(self):
+        return [
+            (form.get_unique_id(), u'{} / {}'.format(form.get_module().default_name(), form.default_name()))
+            for form in self.get_app().get_forms() if form.form_type == "advanced_form"
+        ]
+
+    @staticmethod
+    def _merge_actions(source_actions, extra_actions):
+        new_actions = OrderedDict(
+            (action.case_tag, LoadUpdateAction.wrap(action.to_json()))
+            for action in source_actions.load_update_cases
+        )
+        for action in extra_actions.load_update_cases:
+            new_action = new_actions.get(action.case_tag, LoadUpdateAction(case_tag=action.case_tag))
+            overwrite_properties = [
```

Comment:
You've probably thought about this already, as new properties are added to actions in the future, is it likely that they'll also need to be added to this list? Just wondering whether it would make sense to make the hard-coded list the properties *not* to overwrite.